pkg/rebuild/schema/schema.go

```diff
@@ -197,112 +140,28 @@ type SmoketestResponse struct {
 // RebuildPackageRequest is a single request to the rebuild package endpoint.
 type RebuildPackageRequest struct {
 	// TODO: Should this also include Artifact?
-	Ecosystem        rebuild.Ecosystem
-	Package          string
-	Version          string
-	ID               string
-	StrategyFromRepo bool
+	Ecosystem        rebuild.Ecosystem `form:",required"`
+	Package          string            `form:",required"`
+	Version          string            `form:""`
+	ID               string            `form:",required"`
+	StrategyFromRepo bool              `form:""`
 }
 
 var _ Message = &SmoketestRequest{}
 
-// FromValues parses the rebuild form values into a RebuildPackageRequest.
-func (req *RebuildPackageRequest) FromValues(form url.Values) error {
-	// TODO: check that it's a valid ecosystem?
-	req.Ecosystem = rebuild.Ecosystem(form.Get("ecosystem"))
-	if req.Ecosystem == "" {
-		return errors.New("No ecosystem provided")
-	}
-	req.Package = form.Get("pkg")
-	if req.Package == "" {
-		return errors.New("No pkg provided")
-	}
-	version := form.Get("version")
-	if version != "" {
-		req.Version = version
-	}
-	req.ID = form.Get("id")
-	if req.ID == "" {
-		return errors.New("No ID provided")
-	}
-	if fromRepo := form.Get("strategyFromRepo"); fromRepo != "" {
-		strategyFromRepo, err := strconv.ParseBool(fromRepo)
-		if err != nil {
-			return err
-		}
-		req.StrategyFromRepo = strategyFromRepo
-	}
-	return nil
-}
-
-// ToValues converts a RebuildPackageRequest into a url.Values.
-func (req RebuildPackageRequest) ToValues() (url.Values, error) {
-	vals := url.Values{}
-	vals.Set("ecosystem", string(req.Ecosystem))
-	vals.Set("pkg", req.Package)
-	vals.Set("version", req.Version)
-	vals.Set("id", req.ID)
-	vals.Set("strategyFromRepo", strconv.FormatBool(req.StrategyFromRepo))
-	return vals, nil
-}
-
 // InferenceRequest is a single request to the inference endpoint.
 type InferenceRequest struct {
-	Ecosystem    rebuild.Ecosystem
-	Package      string
-	Version      string
-	LocationHint *rebuild.LocationHint
```

Comment:
Can we keep this limited to LocationHint for now? IIUC we don't have any guardrails to detect unsupported Strategy types in inference, and I don't think we have near-term plans to support other strategy types either.

Reply:
Okay so I've left the logic in and have hooked it back in to the parsing logic in #47